app/models/concerns/media_facebook_item.rb

```diff
@@ -175,10 +93,23 @@ def parse_from_facebook_html
 
   def get_facebook_info_from_metadata
     get_metatags(self)
-    metadata = self.get_opengraph_metadata
-    self.data['author_name'] = metadata['title']
-    self.data['text'] = metadata['description']
-    self.data['photos'] = metadata['picture'].nil? ? [] : [metadata['picture']]
+    metadata = self.get_opengraph_metadata || {}
```

Comment:
isn't there another parser for OG tags? do we need to replicate its work here?

Reply:
I'm reusing the same OG parser we use for the generic page parser: https://github.com/meedan/pender/blob/develop/app/models/concerns/media_page_item.rb#L48

Reply:
good - I meant: is the generic page parser always invoked even for Fb links? if so, we should not need to re-invoke it here.

Reply:
no it's not... I'm just reusing the method but it's called only once (which is here at this line)

Reply:
weird... I thought we had some parsers that applied to all links.. let's discuss too

Reply:
no, we have steps that apply to all links (for example, archiving, extract metatags (which are reused for the OG extracting), extract JSON-LD, extract Schema.org tags, etc.)

Reply:
that's what I was asking: if extract metatags is always called, do we need to call it again in the Fb step?

Reply:
we call it here just to map the metatags to field values, but we extract only once

Reply:
I see thanks